internal/services/network/virtual_hub_routing_intent_resource_test.go

```diff
@@ -0,0 +1,183 @@
+package network_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2023-04-01/virtualwans"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type VirtualHubRoutingIntentResource struct{}
+
+func TestAccVirtualHubRoutingIntent_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_routing_intent", "test")
+	r := VirtualHubRoutingIntentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccVirtualHubRoutingIntent_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_routing_intent", "test")
+	r := VirtualHubRoutingIntentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccVirtualHubRoutingIntent_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_routing_intent", "test")
+	r := VirtualHubRoutingIntentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
```

Comment:
can we add another test step to change the values within the `routing_policy`s to confirm that updating the blocks works as expected?

Reply:
Updated